fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java

```diff
@@ -127,37 +126,81 @@ public static void init(String[] args) throws UnknownHostException {
             System.exit(-1);
         }
         fileStoredHostType = prop.getProperty(HOST_TYPE, null);
+        // Check if the ROLE file has property 'hostType'
+        // If it not has property 'hostType', start with IP
+        // If it has property 'hostType' & hostType = IP, start with IP
         if (null == fileStoredHostType || fileStoredHostType.equals(HostType.IP.toString())) {
             initAddrUseIp(hosts);
             return;
         }
+        // If it has property 'hostType' & hostType = FQDN, start with FQDN
         initAddrUseFqdn(hosts);
     }
 
     @VisibleForTesting
-    static void initAddrUseFqdn(List<InetAddress> hosts) throws UnknownHostException {
+    static void initAddrUseFqdn(List<InetAddress> hosts) {
         useFqdn = true;
-        InetAddress uncheckedLocalAddr = InetAddress.getLocalHost();
-        if (null == uncheckedLocalAddr) {
-            LOG.error("get a null localhost when start fe use fqdn");
+
+        // Try to get FQDN from host
+        Process hostname = null;
+        try {
+            hostname = Runtime.getRuntime().exec("hostname -f");
```

Comment:
I suggest you call the function from java build-in library instead of depending on the result of some commands in shell. 

```java
InetAddress.getLocalHost().getCanonicalHostName()
```

1. We can not assume the behavior of this command in the shell/OS of every user's environment.
2. Even if we could, there's still a chance that this command would fail, but the reason will be hard to guess since we have captured neither the stdout/stderr of this command nor have we captured the return code. It would be much easier if we just call a function and print exception stack when fail.
3. We can also avoid the complicated code on parsing the result from the output of this execution.

Reply:
done.